### PR TITLE
Up H4s to H3 in FAQ

### PR DESCRIPTION
**Pull request by adamboros**
With H4s the questions are not showing up in the right side TOC. There seems to be no reason to keep these H4 either.

### Security

- [X] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

**Comment by s-santillan** [comment]
I'm torn. Hierarchically, it's correct that after h2 is h3, not h4. But I look at the sidebar and wonder why questions have to be a header at all — there's a lot of clutter on the sidebar, wouldn't the user be better off skimming the main body? I'm not sure if I'm making a mountain out of a molehill or if this is even a hill to bring up.

I am requesting a review from @pabloest for an alternate perspective.

**Comment by pabloest** [comment]
The right-side TOC column seems too narrow to make it an easy-to-skim TOC mostly due to the length of the questions (and partially a function of the window size).

For this page we could consider hiding the right-side TOC ([see here](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#hide_table_of_contents) about adding `hide_table_of_contents` to a doc’s frontmatter) and instead using an inline TOC using the `toc` variable [per this doc](https://docusaurus.io/docs/markdown-features/inline-toc).

**Comment by s-santillan** [comment]
@pabloest Your alternative sounds good to me; what do you think @adamboros ?

**Comment by adamboros** [comment]
Sounds good to me, but we can also just move everything to H4. I didn't expected it to break the sidebar so much and just matched the headers in comparisons. Let me know which one do you prefer and I'll fix it.

**Comment by s-santillan** [comment]
Hi @adamboros I think it was right for you to make the headers be semantically correct (from h4 to h3). I'm ok with forgoing the sidebar for this doc because readability/usability is most important in design.  Therfore, I think @pabloest has the right solution in having an inline toc. 😄

**Comment by adamboros** [comment]
image

**Comment by pabloest** [comment]
Hi @adamkvitek for just a few questions + answers, can you propose how you would change them?

File: docs/faq.md [hunk 1]
```diff
@@ -1,39 +1,43 @@
...
 
 The [Semgrep command line tool](https://github.com/returntocorp/semgrep) is open-source, licensed under [GPL v3](<https://tldrlegal.com/license/gnu-general-public-license-v3-(gpl-3)>) — you can use it at work, on private and proprietary code, no problem!
 
 The [Semgrep Registry](https://semgrep.dev/explore) contains rules from different contributors. Most rules, including all community-contributed rules, are under [MIT license](https://tldrlegal.com/license/mit-license). Some of the Registry rules written by r2c are licensed under the [Semgrep Registry license](../licensing/#semgrep-registry-license).
 
 The source for many r2c-written Registry rules is available at [returntocorp/semgrep-rules](https://github.com/returntocorp/semgrep-rules/). Those licensed under the Semgrep Registry license can’t be resold without r2c’s permission. Since r2c offers a paid, hosted application (Semgrep App), it’s important to have this restriction so other companies, like major cloud providers, can’t just resell our rules as a competing service.
 
-#### Is it ok to run Semgrep or the r2c rules on my work projects?
+### Is it ok to run Semgrep or the r2c rules on my work projects?
```
Comment (adamkvitek) [inline docs/faq.md:28]:
```suggestion
### It's ok and safe to run Semgrep or the r2c rules on my work projects
```

File: docs/faq.md [hunk 2]
```diff
@@ -1,39 +1,43 @@
...
 
 Yes! Semgrep is safe to run on your private code. The [Semgrep Registry license’s](../licensing/#semgrep-registry-license) commercial restrictions only come into effect if you are _selling_ the rules provided in the semgrep-rules repository. If that’s the case, you’ll need to talk with r2c first to get permission.
 
-#### I’m a security professional and want to use the semgrep-rules project with my clients as part of my paid services. Is that ok?
+### I’m a security professional and want to use the semgrep-rules project with my clients as part of my paid services. Is that ok?
 
 If you have a typical consulting service and running `semgrep` or `semgrep-rules` is part of your assessments, that’s great and of course feel free to refer your clients to the [Semgrep App](https://semgrep.dev/). But if your entire service is about scanning code and you want to charge for running the `semgrep-rules` repository that r2c and the community created and maintains, you need to discuss this with r2c.
 
-#### What is your support policy?
+### What is your support policy?
```
Comment (adamkvitek) [inline docs/faq.md:36]:
```suggestion
### Understanding r2c support policy
```

File: docs/faq.md [hunk 3]
```diff
@@ -1,39 +1,43 @@
...
 
 Help is available for all users, free or otherwise, through the [r2c Community Slack](https://r2c.dev/slack). Semgrep Team tier customers receive 8\*5 email/phone/Slack support with committed SLAs. See [Support](../support/) for more details.
 
-#### Can I embed the Playground in my website or blog post?
+### Can I embed the Playground in my website or blog post?
```
Comment (adamkvitek) [inline docs/faq.md:40]:
```suggestion
### Embedding the Playground in website or blog post
```

File: docs/faq.md [hunk 4]
```diff
@@ -1,39 +1,43 @@
...
 
 Yes! You can embed a special version of Semgrep Playground with an `iframe`. The source is `https://semgrep.dev/embed/editor?snippet=<snippet-id>` where the `snippet-id` is either the short identifier generated when you share a Playground link (this usually looks like `DzKv`) or the named identifier from a saved rule (this usually looks like `username:rule-name`).
```
Comment (adamkvitek) [inline docs/faq.md:42]:
```suggestion
To embed the Playground in a website or a blog post using a special version of Semgrep Playground with an iframe. The source is `https://semgrep.dev/embed/editor?snippet=<snippet-id>` where the `snippet-id` is either the short identifier generated when you share a Playground link (this usually looks like `DzKv`) or the named identifier from a saved rule (this usually looks like `username:rule-name`).
```

File: docs/faq.md [hunk 5]
```diff
@@ -117,25 +121,25 @@ See [the Semgrep development philosophy](../contributing/semgrep-philosophy/) fo
 
 ## Privacy and Security
 
-#### Where do you store data?
+### Where do you store data?
```
Comment (adamkvitek) [inline docs/faq.md:124]:
```suggestion
### Where does r2c store data?
```

Maybe we can have question here. But one which shows whom are we asking.

**Comment by adamkvitek** [comment]
> Hi @adamkvitek for just a few questions + answers, can you propose how you would change them?

@pabloest Made a few comments - https://github.com/returntocorp/semgrep-docs/pull/394#discussion_r812323368

Generally, I am not very happy with the state of these FAQs. Can I please review them thoroughly? This is not out of pure perfectionism. I just see that the things here can be rewritten and grouped together in more logical chunks maybe. I just had a quick five minutes look now, spending half an hour might be worthwhile.

**Comment by pabloest** [comment]
Thanks for doing that, @adamkvitek! OK, why don’t you try that on a separate branch, that way @adamboros can merge this one about the headers and TOC and you can address the FAQ thing separately?

**Comment by adamkvitek** [comment]
@pabloest good idea! Yes, let's merge this, I can work on this on a different branch. ;-)

**Pull request merged**